tests/unit/games-dao.test.js

```diff
@@ -0,0 +1,215 @@
+    serializeGamesStub.returnsArg(0);
+    serializeGameStub.returnsArg(0);
+
+    gamesDao = proxyquire(`${appRoot}/api/v1/db/oracledb/game-dao`,
+      {
+        [`${appRoot}/api/v1/serializers/games-serializer`]: {
+          serializeGame: serializeGameStub,
+          serializeGames: serializeGamesStub,
+        },
+      });
+  });
+
+  afterEach(() => sinon.restore());
+
+  describe('Test getGames', () => {
+    const validTestCases = [
+      testCases.emptyResult,
+      testCases.multiResult,
+      testCases.singleResult,
+    ];
+    _.forEach(validTestCases, (testCase) => {
+      it(`getGames should be fulfilled with ${testCase.description}`, () => {
+        createConnStub(testCase);
+
+        const expectedResult = testCase.data.rows;
+        const result = gamesDao.getGames();
+        return result.should
+          .eventually.be.fulfilled
+          .and.deep.equal(expectedResult);
+      });
+    });
+  });
+
+  describe('Test getGameById', () => {
+    const validTestCases = [
+      testCases.emptyResult,
+      testCases.singleResult,
+    ];
+    _.forEach(validTestCases, (testCase) => {
+      it(`getGameById should be fulfilled with ${testCase.description}`, () => {
+        createConnStub(testCase);
+
+        const expectedResult = testCase.data.rows[0] ? testCase.data.rows : undefined;
+        const result = gamesDao.getGameById(fakeId);
+        return result.should
+          .eventually.be.fulfilled
+          .and.deep.equal(expectedResult);
+      });
+    });
+
+    it('getGameById should reject with multiple results', () => {
+      const testCase = testCases.multiGameResult;
+
+      const error = 'Expect a single object but got multiple results.';
+
+      createConnStub(testCase);
+
+      const result = gamesDao.getGameById(fakeId);
+      return result.should.eventually.be.rejectedWith(Error, error);
+    });
+  });
+
+  describe('Test postGame', () => {
+    it('Improper body should be rejected', () => {
+      createConnStub();
+
+      const result = gamesDao.postGame({});
+      return result.should
+        .eventually.be.rejected
+        .and.be.an.instanceOf(TypeError);
+    });
+
+    it('Should be rejected when there are no OutId in the result field', () => {
+      createConnStub(testCases.singleResult);
+
+      const result = gamesDao.postGame(fakeGamePostBody);
+      return result.should
+        .eventually.be.rejected
+        .and.be.an.instanceOf(TypeError);
+    });
+
+    it('Should be fulfilled with single result', () => {
+      const expectedResult = [{}];
+      createConnStub(testCases.singleResultWithOutId);
+
+      const result = gamesDao.postGame(fakeGamePostBody);
+      return result.should
+        .eventually.be.fulfilled
+        .and.deep.equal(expectedResult);
+    });
+  });
+
+  describe('Test validateGame', () => {
+    it('The function should be rejected when the database response does not contain COUNT(1) field', () => {
```

Comment:
These 3 `COUNT(1)` tests can be done in a loop. Just make a list containing references to the different functions being tested.